### PR TITLE
Entry for CPP Math Function ` atan2()`

### LINKED ISSUE #952: [Term Entry] C++ Math Functions: `atan2()`
### Reason/inspiration (optional)

Under `cpp/concepts/math-functions/terms/`, create a new term entry for the `atan2()` function.

- In the `terms/` folder, make a new folder called `atan2`
- Inside the `atan2` folder, create a new markdown file called `atan2.md`
- Inside the file, be sure to include a "Syntax" section with a \```pseudo code snippet.
- Let's include an "Example" section with a \```cpp code snippet.
- Lastly, let's include a "Codebyte Example" with a runnable \```codebyte/cpp

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by shlok-2003**
### Description

Made an initial CPP Math Function atan2() file. Closes #952

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by shlok-2003** [comment]
Hey @Dusch4593, I have made the initial file. Please review it

File: content/cpp/concepts/math-functions/terms/atan2/atan2.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+The **`atan2()`** function returns the inverse tangent of a coordinate (y/x), where y is the y-coordinate value and x is the x-coordinate value.
+
+## Syntax
+
+```pseudo
+atan2(y, x)
+```
+
+The `y`, `x` parameter can be one of the following [data types](https://www.codecademy.com/resources/docs/java/data-types)
```
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/atan2/atan2.md:22]: This needs to mention the data types.

File: content/cpp/concepts/math-functions/terms/atan2/atan2.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+#include <cmath>
+
+int main() {
+    // Declaration of the two parameters
+    double x = 10.0, y = 20.0;
+
+    // Output
+    std :: cout << atan2(y, x) << std :: endl;
+
+    1.10715
```
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/atan2/atan2.md:39]:
```suggestion
```

File: content/cpp/concepts/math-functions/terms/atan2/atan2.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+#include <iostream>
+#include <cmath>
+
+int main() {
+    // Declaration of the two parameters
+    double x = 10.0, y = 20.0;
+
+    // Output
+    std :: cout << atan2(y, x) << std :: endl;
+
```
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/atan2/atan2.md:38]:
```suggestion
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
I did a first review and left a couple of comments.

File: content/cpp/concepts/math-functions/terms/atan2/atan2.md (hunk 1 above)
Comment (SSwiniarski) [reply content/cpp/concepts/math-functions/terms/atan2/atan2.md]: @shlok-2003 You should list what data types x and y can be. It says "the following data types" but does not list the data types following the sentence.

Comment (shlok-2003) [reply content/cpp/concepts/math-functions/terms/atan2/atan2.md]: I have added the data type list in the file

**Review by SSwiniarski — APPROVED** [review]
This looks good for a second review

File: content/cpp/concepts/math-functions/terms/atan2/atan2.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Functions'
+  - 'Arithmetic'
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
+The **`atan2()`** function returns the inverse tangent of a coordinate (y/x), where y is the y-coordinate value and x is the x-coordinate value.
```
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/atan2/atan2.md:14]:
```suggestion
The **`atan2()`** function returns the inverse tangent of the quotient of a pair of coordinates (y/x), in radians.
```

File: content/cpp/concepts/math-functions/terms/atan2/atan2.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+## Syntax
+
+```pseudo
+atan2(y, x)
+```
+
+The `y`, `x` parameter can be one of `double`, `float` or `long double` datatype.
```
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/atan2/atan2.md:22]:
```suggestion
The first and second parameters, `y` and `x`, can be one of `double`, `float` or `long double` [data type](https://www.codecademy.com/resources/docs/cpp/data-types).
```

Comment (Dusch4593) [reply content/cpp/concepts/math-functions/terms/atan2/atan2.md:22]: Can we also include 1-3 sentences about how the `atan2()` function behaves when either parameter is negative, or if both parameters are equal to zero (0)?

File: content/cpp/concepts/math-functions/terms/atan2/atan2.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+#include <iostream>
+#include <cmath>
+
+int main() {
+    // Declaration of the two parameters
+    double x = 10.0, y = 20.0;
+
+    // Output
+    std :: cout << atan2(y, x) << std :: endl;
+}
```
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/atan2/atan2.md:38]:
```suggestion
int main() {
  double x = 10.0, y = 20.0;
  
  std::cout << atan2(y, x) << std::endl;
}
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
Hey @shlok-2003, I've finished my review and left you some comments/suggestions. Additionally, could we make a new "Codebyte Example" section of the end of the file, that features a runnable \```codebyte/cpp block?

**Comment by shlok-2003** [comment]
Hey @Dusch4593, I have made the necessary changes and added new codebyte example

**Comment by shlok-2003** [comment]
Hey @Dusch4593, are there any updates?

File: content/cpp/concepts/math-functions/terms/atan2/atan2.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+## Syntax
+
+```pseudo
+atan2(y, x)
+```
+
+- The first and second parameters, `y` and `x`, can be one of `double`, `float` or `long double` [data type](https://www.codecademy.com/resources/docs/cpp/data-types).
+- The return value has the range of [-π, π].
+- The function returns negative value when first parameter is negative.
+- If both the parameters are 0, then the value is 0.
```
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/atan2/atan2.md:25]:
```suggestion
The [data type](https://www.codecademy.com/resources/docs/cpp/data-types) of the `y` and `x` parameters can be `double`, `float` or `long double`.

- The return value has the range of [-π, π].
- A negative value is returned if the first parameter is negative.
- If both parameters are 0, then the value is 0.
```

File: content/cpp/concepts/math-functions/terms/atan2/atan2.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+```codebyte/cpp
+#include <iostream>
+#include <cmath>
+int main()
+{
+  double x, y;
+  cin >> x >> y;
+
+  std :: cout << atan2(y, x) << std :: endl;
+  return 0;
```
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/atan2/atan2.md:62]: We need to rewrite this example and omit `cin`. Unfortunately, input/output function don't work properly in Codebytes.

Maybe this example could feature one of the edge cases mentioned in the "Syntax" section?

File: content/cpp/concepts/math-functions/terms/atan2/atan2.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Functions'
+  - 'Arithmetic'
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
+The **`atan2()`** function returns the inverse tangent of the quotient of a pair of coordinates (y/x), in radians.
```
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/atan2/atan2.md:14]:
```suggestion
The **`atan2()`** function returns the inverse tangent of the quotient of a coordinate pair (y/x), in radians.
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@shlok-2003 Finished with my review and left some comments/suggestions.

File: content/cpp/concepts/math-functions/terms/atan2/atan2.md (hunk 8 above)
Comment (shlok-2003) [reply content/cpp/concepts/math-functions/terms/atan2/atan2.md:62]: Hey @Dusch4593, I am mentioning the case where both the coordinates are zero.

**Comment by shlok-2003** [comment]
Hey @Dusch4593, the file has been updated 👍🏻

**Review by Dusch4593 — APPROVED** [review]
LGTM, @shlok-2003! 👍🏻

Before we merge, I made some edit in 52fb54b6180f7f5da7f45cb153353f6fe874bda0 that I wanted to note:

- Edited `Description` on line # 3 to match opening sentence on # 14 more closely.
- Added oxford comma on line # 22.
- Opted for a more direct reference to the parameter on line # 25 (from "first" to "y").
- Line # 30, appended some text about the edge case in the example.
- Removed extra spacing between C++ code and comment on line # 41.
- Line # 52: Added return value to \```shell snippet from second call to `atan2()`.
- Lin3 # 57: Removed text about user input (since it was removed) and replaced with mention about negative coordinate value.

cc: @SSwiniarski

**Comment by Dusch4593** [comment]
Congrats and thank you for another solid entry to Codecademy Docs, @shlok-2003! 🎊 🎉 🥳

The entry should now be live: https://www.codecademy.com/resources/docs/cpp/math-functions/atan2

If you'd like to make your Codecademy profile visible on the entry page, you'll need to add your GitHub username to your Codecademy profile, log out of Codecademy, and log back in through your GitHub account:

Screen Shot 2022-11-30 at 12 17 03 PM

Looking forward to your next Docs contribution! 😄

**Pull request merged**